src/Compilers/CSharp/Portable/Binder/Binder_Invocation.cs

```diff
@@ -1439,7 +1439,7 @@ private BoundExpression GetDefaultParameterSpecialNoConversion(SyntaxNode syntax
                 }
                 else
                 {
-                    if (Compilation.SourceModule != parameters[paramsIndex].ContainingModule)
+                    if (Compilation.SourceModule != parameters[paramsIndex].ContainingModule || parameters[paramsIndex].ContainingType.IsAnonymousType)
```

Comment:
This doesn't seem like a relevant addition to the check to me. It won't catch usages that don't involve actually invoking the params; we already have a lang version error on declarations that use `params` within the compilation, so we should be warning on the location that actually creates the anonymous type. Otherwise, we're missing scenarios.

Reply:
This check is not meant to catch non-invocation scenarios and it adds an observable benefit for scenarios that it targets.

Reply:
It does feel like this would add more extra positives than looking at the conversion, but I'm not generally against it. I'm just not sure it's necessary if we error on the conversion instead.